### PR TITLE
Caching map results with incremental computation

### PR DESCRIPTION
**Pull request by dharini-s**
[Issue #373]

- Modified common module to interact with mapresultcache module.
- Created mapresultcache module that caches map results and performs incremental computation.
- Added library hashmap.js.

### CODE REVIEW

File: examples/twittermap/web/public/javascripts/lib/hashmap.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/**
```
Comment (JavierJia) [inline examples/twittermap/web/public/javascripts/lib/hashmap.js]: we should remove this file.

Comment (waans11) [reply examples/twittermap/web/public/javascripts/lib/hashmap.js]: Got it. How can I make this happen? Check out her branch and remove this file and then push the change to her branch? Is this possible?

Comment (waans11) [reply examples/twittermap/web/public/javascripts/lib/hashmap.js]: I will request a new PR.

**Comment by waans11** [comment]
Since a file needs to be deleted, I have uploaded a new patch and merged it. So, I am going to close this pull request.

**Pull request closed without merging**